doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
+>>> g.add_vertices(5)
+
+This changes the graph `g` in place. You can specify the name of the vertices if you wish.
+
+To remove nodes, use :meth:`Graph.delete_vertices`:
+
+>>> g.delete_vertices(None)  # remove all vertices
+>>> g.delete_vertices([1, 2])
+
+Again, you can specify the names or the actual :class:`Vertex` objects instead.
+
+To add edges, use :meth:`Graph.add_edge` and :meth:`Graph.add_edges`:
+
+>>> g.add_edge(0, 2)
+>>> g.add_edges([(0, 2), (1, 3)])
+
+To remove edges, use :meth:`Graph.delete_edges`:
+
+>>> g.delete_edges(None)   # remove all edges
+>>> g.delete_edges([0, 5]) # remove by edge ID
+
+You can also remove edges between source and target nodes.
+
+To contract vertices, use :meth:`Graph.contract_vertices`. Edges between contracted vertices will become loops.
+
+Graph operators
++++++++++++++++++
+It is possible to compute the union, intersection, difference, and other set operations (operators) between graphs.
+
+To compute the union of the graphs (nodes/edges in either are kept):
+
+>>> gu = ig.union([g, g2, g3])
+
+Similarly for the intersection (nodes/edges present in all are kept):
+
+>>> gu = ig.intersection([g, g2, g3])
+
+These two operations preserve attributes and can be performed with a few variations. The most important one is that vertices can be matched across the graphs by id (number) or by name.
+
+These and other operations are also available as methods of the :class:`Graph` class:
+
+>>> g.union(g2)
+>>> g.intersection(g2)
+>>> g.disjoint_union(g2)
+>>> g.difference(g2)
+>>> g.complementer()  # complement graph, same nodes but missing edges
+
+and even as numerical operators:
+
+>>> g |= g2
+>>> g_intersection = g and g2
+
+Topological sorting
++++++++++++++++++++
+To sort a graph topologically, use :meth:`Graph.topological_sorting`:
+
+>>> g = ig.Graph.Tree(10, 2, mode=ig.TREE_OUT)
+>>> g.topological_sorting()
+
+Graph traversing
++++++++++++++++++++++
+A common operation is traversing the graph. |igraph| currently exposes breath-first search (BFS) via :meth:`Graph.bfs` and :meth:`Graph.bfsiter`:
+
+>>> [vertices, layers, parents] = g.bfs()
+>>> it = g.bfsiter()  # Lazy version
+
+A depth-first search function is in the works.
+
+To perform a random walk from a certain vertex, use :meth:`Graph.random_walk`:
+
+>>> vids = g.random_walk(0, 3)
+
+Pathfinding and cuts
+++++++++++++++++++++
+Several pathfinding algorithms are available:
+
+- :meth:`Graph.shortest_paths` or :meth:`Graph.get_shortest_paths`
+- :meth:`Graph.get_all_shortest_paths`
+- :meth:`Graph.get_all_simple_paths`
+- :meth:`Graph.spanning_tree` finds a minimum spanning tree
+
+As well as functions related to cuts and paths:
+
+- :meth:`Graph.mincut` calculates the minimum cut between the source and target vertices
+- :meth:`Graph.st_mincut` - as previous one, but returns a simpler data structure
+- :meth:`Graph.mincut_value` - as previous one, but returns only the value
+- :meth:`Graph.all_st_cuts`
+- :meth:`Graph.all_st_mincuts`
```

Comment:
Algorithmically, these are related to flows, not to shortest paths. I don't think it's a bad fit here, but I wonder if it makes sense to be a bit redundant and mention things more than once when they are related to multiple areas.

Reply:
yep added